### PR TITLE
Part I: An admin user can filter products by uploaded CSV

### PR DESCRIPTION
**Pull request by machikoyasuda**
## User stories
- A user should be able to click Filter by File and see a card with an Import CSV button, and text.
- A user should be able to click X and close the card.
- A user should be able to upload a CSV file, and see the returned products.
- A user should be able to see an InlineAlert reporting what number of products were found and not found.
- A user should be able to see the total number of products found, and total number of products selected.
- A user should be able to see an InlineAlert with an error if no products were found. The table should then show all products again.

## Changes
- Upgrades dropzone to v10, to React Hooks

## Solution
- [x] - Add Filter by File button to Actions dropdown, with i18n'd copy.
- [x] - Add a Quick Edit Card
- [x] - Add X to the card to close the card
- [x] - Add a Catalyst button with the Upload option
- [x] - The Create Product button should disappear when the Card is there.
- [x] - The table should show `All Products` with the number of products
- [x] - When products are selected, show the number of selected products

### From Reaction: Action dropdown
_This component is in Reaction, not Catalyst **yet**._
This dropdown is composed of Material-UI Buttons and Catalyst ConfirmDialogs.
- [x] Hover color is wrong.
- [x] "Actions" should show up after you've opened it
- [x] "Actions" should change color, remove border and hide the arrow once you've clicked it

Screen Shot 2019-08-12 at 9 43 08 PM
Screen Shot 2019-08-12 at 9 42 46 PM
Screen Shot 2019-08-12 at 9 42 50 PM

### From Reaction x MUI: Quick Edit Card
_This component is in Reaction, not Catalyst **yet**._
This component is composed of Material-UI Card, CardHeader, CardContent w. Reaction styling.

Screen Shot 2019-08-12 at 9 43 34 PM

Screen Shot 2019-08-13 at 10 02 33 AM

### From Catalyst: Chips

Once Chips are updated in Catalyst, they'll be updated in this ticket w/ correct styles.

### Testing

1. Test a user filters products w 1 file
1. Test a user filters products w multiple files
1. Test a user filters products and none are found
1. Test a user filters products and some are found, but not all
1. Test a user filters products and selects/deselects some of them

### CODE REVIEW

File: imports/plugins/included/product-variant/components/productGrid.js [hunk 1]
```diff
@@ -21,35 +21,71 @@ import TableRow from "@material-ui/core/TableRow";
...
+    "&:hover": {
+      backgroundColor: "#EBF7FC"
+    }
   }
 });
 
 class ProductGrid extends Component {
   static propTypes = {
     classes: PropTypes.object,
+    files: PropTypes.arrayOf(PropTypes.file),
```
Comment (machikoyasuda) [inline imports/plugins/included/product-variant/components/productGrid.js]: @mikemurray Should we make a custom `PropTypes.file` type? Getting this error: ```Failed prop type: Property `files` of component `ProductGrid` has invalid PropType notation inside arrayOf.```

Comment (machikoyasuda) [reply imports/plugins/included/product-variant/components/productGrid.js]: Changed it to `PropTypes.object`

File: imports/plugins/included/product-variant/components/productGrid.js [hunk 2]
```diff
@@ -96,23 +132,35 @@ class ProductGrid extends Component {
...
     if (filterByProductIds) {
       return (
         <div className={classes.filterCountContainer}>
-          <Typography variant="h4" display="inline">
+          <Typography variant="h4" display="inline" className={classes.productsTitle}>
             {i18next.t("admin.productTable.bulkActions.filteredProducts")}
           </Typography>
           <Typography variant="h5" display="inline" className={classes.filterCountText}>
-            {i18next.t("admin.productTable.bulkActions.selectedCount", { count })}
+            {i18next.t("admin.productTable.bulkActions.selectedCount", { selectedCount })}
```
[the remark this answers is not in the document]
Comment (machikoyasuda) [reply imports/plugins/included/product-variant/components/productGrid.js]: the selected copy is always “1 selected” or “12 selected” so there’s no pluralization required

[the remark this answers is not in the document]
Comment (mikemurray) [reply imports/plugins/included/product-variant/components/productGrid.js]: The number is still missing though.

Products

[the remark this answers is not in the document]
Comment (machikoyasuda) [reply imports/plugins/included/product-variant/components/productGrid.js]: resolved in this latest commit

**Pull request merged**